homeassistant/components/agent_dvr/camera.py

```diff
@@ -0,0 +1,263 @@
+"""Support for Agent camera streaming."""
+from datetime import timedelta
+import logging
+
+from agent import AgentError
+import voluptuous as vol
+
+from homeassistant.components.camera import (
+    STATE_IDLE,
+    STATE_RECORDING,
+    STATE_STREAMING,
+    SUPPORT_ON_OFF,
+    SUPPORT_STREAM,
+)
+from homeassistant.components.mjpeg.camera import (
+    CONF_MJPEG_URL,
+    CONF_STILL_IMAGE_URL,
+    MjpegCamera,
+    filter_urllib3_logging,
+)
+from homeassistant.const import ATTR_ATTRIBUTION, CONF_NAME
+from homeassistant.helpers import entity_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import (
+    ATTRIBUTION,
+    CAMERA_SCAN_INTERVAL_SECS,
+    CONNECTION,
+    DOMAIN as AGENT_DOMAIN,
+)
+from .helpers import service_signal
+
+SCAN_INTERVAL = timedelta(seconds=CAMERA_SCAN_INTERVAL_SECS)
+
+_LOGGER = logging.getLogger(__name__)
+
+_DEV_EN_ALT = "enable_alerts"
+_DEV_DS_ALT = "disable_alerts"
+_DEV_EN_REC = "start_recording"
+_DEV_DS_REC = "stop_recording"
+_DEV_SNAP = "snapshot"
+
+CAMERA_SERVICES = {
+    _DEV_EN_ALT: "async_enable_alerts",
+    _DEV_DS_ALT: "async_disable_alerts",
+    _DEV_EN_REC: "async_start_recording",
+    _DEV_DS_REC: "async_stop_recording",
+    _DEV_SNAP: "async_snapshot",
+}
+
+
+async def async_setup_entry(
+    hass, config_entry, async_add_entities, discovery_info=None
+):
+    """Set up the Agent cameras."""
+    filter_urllib3_logging()
+    cameras = []
+
+    server = hass.data[AGENT_DOMAIN][config_entry.entry_id][CONNECTION]
+    server.cameras = []
+    if not server.deviceList:
+        _LOGGER.warning("Could not fetch cameras from Agent server")
+        return
+
+    for device in server.deviceList:
+        if device.typeID == 2:
+            camera = AgentCamera(device)
+            cameras.append(camera)
+            server.cameras.append(camera)
+
+    async_add_entities(cameras)
+
+    platform = entity_platform.current_platform.get()
+    for service, method in CAMERA_SERVICES.items():
+        platform.async_register_entity_service(
+            service, {vol.Required("entity_id"): str}, method
```

Comment:
```suggestion
            service, {}, method
```

This is going to get passed through `make_entity_service_schema` so `entity_id` will already get added